plugins/pulp_rpm/plugins/migrations/0016_new_yum_distributor.py

```diff
@@ -176,3 +177,58 @@ def _re_publish_repository(repo, distributor):
 def _remove_legacy_publish_dirs():
     _clear_orphaned_publish_dirs(OLD_ROOT_PUBLISH_DIR, OLD_HTTP_PUBLISH_DIR)
     _clear_orphaned_publish_dirs(OLD_ROOT_PUBLISH_DIR, OLD_HTTPS_PUBLISH_DIR)
+
+
+def _working_dir_root():
+    storage_dir = pulp_config.config.get('server', 'storage_dir')
+    dir_root = os.path.join(storage_dir, 'working')
+    return dir_root
+
+
+def _repo_working_dir():
+    dir = os.path.join(_working_dir_root(), 'repos')
+    return dir
+
+
+def repository_working_dir(repo_id, mkdir=True):
+    """
+    Determines the repository's working directory. Individual plugin working
+    directories will be placed under this. If the mkdir argument is set to true,
+    the directory will be created as part of this call.
+    See the module-level docstrings for more information on the directory
+    structure.
+    @param mkdir: if true, this call will create the directory; otherwise the
+                  full path will just be generated
+    @type  mkdir: bool
+    @return: full path on disk
+    @rtype:  str
+    """
+    working_dir = os.path.join(_repo_working_dir(), repo_id)
```

Comment:
Since _repo_working_dir() just does an os.path.join, why not eliminate it and put what it does here?